### PR TITLE
merging new components into main

### PR DESCRIPTION
**Pull request by sun-mota**
# Alaska Airlines Pull Request

## Merging PRs below at once
- [x] #21
- [x] #27 
- [x] #28 

## A separate PR to come later for below
- [ ] #26 
- [ ] #43 
- [ ] #41 
- [ ] #42 

## Summary by Sourcery

Integrate new components auro-menu and auro-checkbox into the project, along with new date input types for auro-input. Update build scripts and dependencies, and enhance documentation and test coverage for these additions.

New Features:
- Introduce new date input types 'month', 'year', and 'fullYear' for the auro-input component.
- Add auro-menu and auro-checkbox components with various features and configurations.

Enhancements:
- Refactor auro-input to use a new registration method for custom elements.
- Update auro-input to handle new date input types with appropriate validation and formatting.

Build:
- Update package dependencies for auro-button, auro-formvalidation, and auro-icon to newer versions.
- Add build scripts for new components like checkbox and menu.

Documentation:
- Add comprehensive documentation for new components auro-menu and auro-checkbox, including usage examples and API details.

Tests:
- Add test cases for new auro-menu and auro-checkbox components to ensure functionality and integration.

## Summary by Sourcery

Integrate new components auro-menu and auro-checkbox into the project, along with new date input types for auro-input. Update build scripts and dependencies, and enhance documentation and test coverage for these additions.

New Features:
- Introduce new date input types 'month', 'year', and 'fullYear' for the auro-input component.
- Add auro-menu and auro-checkbox components with various features and configurations.

Enhancements:
- Refactor auro-input to use a new registration method for custom elements.
- Update auro-input to handle new date input types with appropriate validation and formatting.

Build:
- Update package dependencies for auro-button, auro-formvalidation, and auro-icon to newer versions.
- Add build scripts for new components like checkbox and menu.

Documentation:
- Add comprehensive documentation for new components auro-menu and auro-checkbox, including usage examples and API details.

Tests:
- Add test cases for new auro-menu and auro-checkbox components to ensure functionality and integration.

### CODE REVIEW

File: web-test-runner.config.mjs [hunk 1]
```diff
@@ -9,10 +9,10 @@ export default {
...
   coverageConfig: {
     threshold: {
-      statements: 80,
+      statements: 70,
       branches: 78,
-      functions: 80,
-      lines: 80
+      functions: 70,
+      lines: 70
     }
```
Comment (chrisfalaska) [inline web-test-runner.config.mjs:16]: Did we want to commit these changes, or was this simply for testing purposes?

Comment (sun-mota) [reply web-test-runner.config.mjs:16]: This is the change that was suggested by Jason from the meeting that we had. It's to make the test passes with the low test coverage.

**Review by chrisfalaska — APPROVED** [review]

File: web-test-runner.config.mjs (hunk 1 above)
Comment (sun-mota) [reply web-test-runner.config.mjs:16]: This can  be reverted after https://github.com/AlaskaAirlines/auro-form/issues/47 gets addressed.

**Pull request merged**